ports/esp32s2/common-hal/wifi/Network.c

```diff
@@ -54,3 +54,38 @@ mp_obj_t common_hal_wifi_network_get_country(wifi_network_obj_t *self) {
         // 2 instead of strlen(cstr) as this gives us only the country-code
         return mp_obj_new_str(cstr, 2);
 }
+
+mp_obj_t common_hal_wifi_network_get_authmode(wifi_network_obj_t *self) {
+    char authmode[16];
+    switch (self->record.authmode) {
+        case WIFI_AUTH_OPEN:
+            strcpy(authmode, "OPEN");
+            break;
```

Comment:
This work?

```suggestion
    const char* authmode = "";
    switch (self->record.authmode) {
        case WIFI_AUTH_OPEN:
           authmode = "OPEN";
            break;
```

Reply:
This worked out fine, thank you for the help! I've updated the Pull Request. I'll create new PRs when they are ready for the other enhancements. :)